tests/driver/api.test.ts

```diff
@@ -0,0 +1,633 @@
+            assert.strictEqual(product1.name, 'Product 1');
+        });
+        it('API ops tests Model.$where()', async () => {
+            //Mode.$where()
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 1'});
+            await product1.save();
+            const whereJSexpressionResp = await Product.$where('this.name === "Product 1"').exec()
+            //find command doesn't support   "filter": { "$where": "this.name === \"Product 1\"" }
+            assert.strictEqual(whereJSexpressionResp.length, 0);
+            //----------------//
+        });
+        it('API ops tests Model.aggregate()', async () => {
+            //Model.aggregate()
+            let error: OperationNotSupportedError | null = null;
+            try {
+                const aggregateResp = await Product.aggregate([{$match: {name: 'Product 1'}}]);
+            } catch (err: any) {
+                error = err;
+            }
+            assert.strictEqual(error!.message, 'aggregate() Not Implemented');
+            //----------------//
+        });
+        //TODO
+        it.skip('API ops tests Model.applyDefaults()', async () => {
+            Model.applyDefaults();
+            new Product().applyDefaults({name: 'Product 1', price: 10, isCertified: true, category: 'cat 1'});
+        });
+        //TODO
+        it.skip('API ops tests Model.bulkSave()', async () => {
+            //Skipping /node_modules/mongoose/lib/model.js:3442:74
+            //Model.bulkSave() error:  TypeError: Cannot read properties of undefined (reading 'find')
+            /*try {
+                const product2 = new Product({name: 'Product 2', price: 20, isCertified: true, category: 'cat 2'});
+                const product3 = new Product({name: 'Product 3', price: 30, isCertified: true, category: 'cat 3'});
+                const bulkSaveResp = await Product.bulkSave([product2, product3]);
+            } catch (err:any) {
+                error = err;
+            }
+            assert.strictEqual(error!.message, 'bulkSave() Not Implemented');*/
+        });
+        it('API ops tests Model.bulkWrite()', async () => {
+            let error: OperationNotSupportedError | null = null;
+            try {
+                const product2 = new Product({name: 'Product 2', price: 20, isCertified: true, category: 'cat 2'});
+                const product3 = new Product({name: 'Product 3', price: 30, isCertified: true, category: 'cat 3'});
+                await Product.bulkWrite([{insertOne: {document: product2}}, {insertOne: {document: product3}}]);
+            } catch (err: any) {
+                error = err;
+            }
+            assert.strictEqual(error!.message, 'bulkWrite() Not Implemented');
+        });
+        it('API ops tests Model.castObjectResp()', async () => {
+            const castObjectResp = Product.castObject({
```

Comment:
I wouldn't recommend testing `castObject()`, because `castObject()` doesn't make any database requests. Works the same regardless of the driver layer.

Reply:
Removed with a comment.